### ISSUE TITLE
Setting initial_window_size leads to flow control errors

### ISSUE THREAD

**Issue by olix0r** [carllerche/h2#146]
There are at least two issues here:

1. As described in https://github.com/carllerche/h2/pull/145, `prioritize` configures the send flow control with the local initial window size.  This is incorrect.

2. `recv` does not configure its flow controller with the local window size

```rust
impl<B, P> Recv<B, P>
where
    P: Peer,
{
    pub fn new(config: &Config) -> Self {
        let next_stream_id = if P::is_server() { 1 } else { 2 };

        let mut flow = FlowControl::new();

        // connections always have the default window size, regardless of
        // settings
        flow.inc_window(DEFAULT_INITIAL_WINDOW_SIZE)
            .expect("invalid initial remote window size");
        flow.assign_capacity(DEFAULT_INITIAL_WINDOW_SIZE);

        Recv {
            init_window_sz: config.local_init_window_sz,
```

Surely the receiver's initial window size should actually be configured according to the settings it advertises...

**Comment by seanmonstar** [comment]
> Surely the receiver's initial window size should actually be configured according to the settings it advertises...

It is! The `Recv.flow` is the control window, which is not affected by settings. However, the initial stream window size is recorded in `init_window_sz: config.local_init_window_sz,`.

**Comment by olix0r** [comment]
TIL...

> The connection flow-control window can only be changed using WINDOW_UPDATE frames.

😱

**Comment by carllerche** [comment]
Yeah....

It makes sense when you think about it though because the remote can start sending out data frames before receiving the initial settings frame.

**Comment by olix0r** [comment]
I expected that to be handled like any other window settings change... the setting is only applied once the peer has acknowledged it (and windows may go negative).  Guess not, though. ¯\\\_(ツ)\_/¯

**Comment by seanmonstar** [comment]
I believe this is fixed, since #145 was merged.
